Sources/TuistDependencies/SwiftPackageManager/Utils/PackageInfoMapper.swift

```diff
@@ -455,6 +477,17 @@ extension ProjectDescription.Settings {
         var cxxFlags: [String] = []
         var swiftFlags: [String] = []
 
+        if moduleMap.type != .none {
+            headerSearchPaths.append("$(SRCROOT)/\(target.relativePath.appending(target.relativePublicHeadersPath))")
+        }
+
+        let allDependencies = packageInfo.recursiveTargetDependencies(of: target)
```

Comment:
I think this is causing issues with `Yams` package in tuist dog-fooding -> here, the variable is called `allDependencies` but in reality, we only add header files of *target* dependencies. We should, however, add header search paths for external dependencies as well. For `SwiftGenKit`, we are missing `$(SRCROOT)/../Yams/Sources/CYaml/include` in `HEADER_SEARCH_PATHS` (if you generate the tuist project with the current PR and add this, it will build `SwiftGenKit` - note there are still issues related to the deployment target but that should be dealt with afterwards)

A possible solution is to gather all the external dependencies, get their `PackageInfo` (we should be able to cache this information if we have already processed it), get transitive dependencies for that external target and add that to `allDepenencies`. Since these headers will not be in the current `$(SRCROOT)`, it'll be necessary to add `$(SRCROOT)/../ExternalDependencyName/Sources/ExternalTargetWithHeaders/include`.

Does that make sense @danyf90?

Reply:
Makes sense, addressed in my last commit.
If you approve it I would merge this PR and address any other change in a separate one